ethcore/src/header.rs

```diff
@@ -186,11 +186,12 @@ impl Header {
 	/// Get the hash of the header excluding the seal
 	pub fn bare_hash(&self) -> H256 {
 		let mut hash = self.bare_hash.borrow_mut();
-		match &mut *hash {
+		match hash.deref_mut() {
```

Comment:
> going against the "everything explicit" philosophy

It is inherited from c++'s `->` operator, which acts similarly. Box and Rc are intended to be analogous to unique_ptr and shared_ptr, so it would be a travesty if derefencing were not automatic in this way. It's really necessary, otherwise you could never write things like my_vec.iter()/len(), since those are methods on [T], and not Vec.

Another point against explicit deref: you have to import the trait everywhere you want to use it :)